lib/firebase_storage/firebase_storage.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:firedart/io/access_exporter.dart';
+import 'package:gcloud/storage.dart';
+import 'package:googleapis_auth/auth_io.dart' as gauth;
+
+import '../auth/firebase_auth.dart';
+
+class FirebaseStorage {
+  final Bucket _bucket;
+
+  FirebaseStorage._internal(this._bucket);
+
+  static Future<FirebaseStorage> getBucket(
+      String projectId, String bucketId, FirebaseAuth auth) async {
+    assert(projectId.isNotEmpty, 'Project ID cannot be null');
+    assert(bucketId.isNotEmpty, 'Bucket ID cannot be null');
+    assert(auth != null, 'Auth cannot be null');
+
+    var credentials = gauth.ServiceAccountCredentials.fromJson(
+        auth.serviceAccount.serviceAccountString);
```

Comment:
You should also be able to access FB Storage using a normal account, right?

Reply:
This is your code unfortunately, I'm not sure how to create a credentials var otherwise

Reply:
I see. I thought it would be the same code as in Firestore, but it isn't.

There's a way to create an AccessCredentials object and pass it to an AutoRefreshingClient. That would be an option.

I need to turn off for the night. Let me know if you find anything, otherwise I can try it myself.

Reply:
@cachapa It seems the other methods require oauth consent. You can't just use username and pass, it will return a URI in the console (probably) to open a google auth screen. Then it'll return the refersh client. So I don't know how you want to handle this.

Technically speaking, this is how the signIn methods in FirebaseAuth should work as well, there's supposed to be an oauth conscent screen.

Reply:
@cachapa  Ye I'm stumped here

Reply:
You shouldn't need oauth consent since you're using your own project's authentication and have full control over it. That is, the client is the official client so you can use oauth Password Grant (providing username+password directly in the client).

I'm sorry but I'm very busy these days so it'll take me longer to be able to look at this more closely. It might just be that using the gcloud package isn't the correct solution here.

Reply:
@cachapa I'm confident it is, and I'm confident there is a way, I just don't know about it. You'll need to be the one to finish this unfortunately.

Reply:
@cachapa Perhaps, until you can get to it, and unless you have any other comments, maybe merge this PR? I'm not going to modify this further

Reply:
Sorry, I'm not comfortable merging incomplete PRs.
The master branch should always represent a production-ready snapshot of the code. I'm even considering 

Maybe you could use your own fork in the mean time?

Reply:
Are you able to add commits to this PR at a later date to finish it?

Reply:
Not really. That's a good point actually.
Maybe it'd make more sense to merge the PR in a separate branch, so I could work on it and merge into master when ready.

Is it possible for you to ask to merge to a new branch on my repo? Otherwise I'll create one for you.

Reply:
@cachapa Yes, you'll need to make a new branch first that I can set this PR to

Reply:
Done: https://github.com/cachapa/firedart/tree/firebase-storage

Reply:
Done